CONTRIBUTING.md

```diff
@@ -91,6 +91,9 @@ and additionally the following rules:
 * Avoid "dead code": Don't add disabled code as nobody but you will understand
   why this is not active. Better leave it out and keep in your local git
   repository, either in `git stash` or a temporary "WIP"-commit.
+* Do not add TODO/FIXME references: TODO comments will rot. Use plain comments
+  to explain or link to external issue trackers. Also see
+  https://blog.usejournal.com/stop-writing-code-comments-28fef5272752
```

Comment:
I don't think that including blog posts of random people is good idea : 
1. This is just personal opinion of single person . Which may or may not be accepted by all contributors of this project. This not some law which everyone should obey without doubts.
2. No guarantee for how long this URL will be valid . Also no guarantee that after some time this URL will point to some different article

Reply:
> I don't think that including blog posts of random people is good idea :
> 
>     1. This is just personal opinion of single person . Which may or may not be accepted by all contributors of this project. This not some law which everyone should obey without doubts.

Exactly. This is why I am referencing an article that is not written by me and I ask reviewers of the project if they agree with the rule.

>     2. No guarantee for how long this URL will be valid . Also no guarantee that after some time this URL will point to some different article

Yes, nothing is forever :) Should I include the date when I accessed the blog article? I doubt we can easily include a checksum of the blog content here to ensure it's the very same over time ;)

Reply:
You can include https://wiki.c2.com/?TodoComments which has arguments pro and against todo/fixme notes.

Having a visible TODO, FIXME comment gives a better opportunity for that idea to be implemented, or deleted altogether if necessary, whilst a ticket in the tracker again, loses relevance and is there rotting for years.

While I can understand how a TODO can be an antipattern in the lib/ or a baseclass, I don't agree that we shouldn't allow todos at all.

Reply:
I like https://wiki.c2.com/ but https://wiki.c2.com/?TodoComments as well most of the related pages are applicable to single-person hacking projects it seems. Here we want to have contribution guidelines for  a big community project.

Obviously tickets in ticket trackers can stay there for long but this is my point: At the time of writing people think it's important so they add TODO/FIXME but the next day you do something different and years later that TODO/FIXME is still there, the code is doing no harm and fixing it is just no priority. Making the code less clear to read and confusing other developers is not helpful. Having a ticket that you close after years realizing that the issue actually turned out to be not relevant anymore even though we might all agree that the code is not perfect or not beautiful is just realistic and should be expected.

Would you agree if I say " … especially in lib/ code or a baseclass"?

Reply:
I would agree with: " Avoid using TODO/FIXME especially in lib/ code or a baseclass, as these will likely rot, it is not forbidden, but recommended not to. Whenever an idea needs to be expressed, use plain comments or add a reference to a ticket in the issue tracker those may rot too, but that's also expected and doesn't pollute the code."

Reply:
I much more prefer this phrasing: "recommended not to"

Reply:
Updated along the suggestion of @foursixnine with minor changes